lib/array.js

```diff
@@ -3,66 +3,81 @@
 module.exports = (api) => {
 
   api.metasync.map = (
-    // Asynchronous map
+    // Asynchronous map (iterate parallel)
     items, // incoming array
-    callback, // function to be executed for each value in the array
-    // current - current element being processed in the array
-    // callback - callback for returning value back to function map
+    fn, // function(current, callback)
+    // to be executed for each value in the array
+    //   current - current element being processed in the array
+    //   callback - function(err, accepted)
+    //     err - error or null
+    //     value - mapped item
+    done // on done callback function(err, result)
     //   err - error or null
-    //   value - result
-    done // callback function on done
-    // err - error or null
-    // data - result if !err
+    //   result - array result
   ) => {
-    const result = [];
-    let hadError = false;
+    const len = items.length;
+    const result = new Array(len);
+    let errored = false;
     let count = 0;
     items.forEach((item, index) => {
-      callback(item, (err, value) => {
+      fn(item, (err, value) => {
+        if (errored) return;
         if (err) {
-          hadError = true;
-          return done(err);
+          errored = true;
+          if (done) done(err);
```

Comment:
Nit: I'd prefer this function to be called `onDone` or `callback`. The current name makes conditions like this one misleading.